### PR TITLE
[Product Block Editor]: add `product-linked` feature flag

### LINKED ISSUE #42914: Create `product-linked` feature flag
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

 Before adding any new feature related to the linked products a feature flag is required until the related functionalities get stable.

**ACs**

```[tasklist]
### Tasks
- [x] Create `product-linked` feature flag
```

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/42914

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable and go to the Product Editor app
2. Open the dev console
3. Print the global `wcAdminFeatures` object
4. Confirm the `product-link` property is defined there

Screenshot 2023-12-20 at 15 10 37

5. Go to Tools > WCA Test Helper
6. Go the Features tab
7. Check that the 'product-link' flag is available

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/config/core.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
 		"product-virtual-downloadable": true,
 		"product-external-affiliate": true,
 		"product-grouped": true,
+		"product-linked": true,
```
Comment (mdperez86) [inline plugins/woocommerce/client/admin/config/core.json]: This cannot be enabled by default in `production`, so it must be `false` here. And set to `true` in `development` `/woocommerce/client/admin/config/development.json`

**Review by mdperez86 — CHANGES_REQUESTED** [review]
Nice job @retrofox! I just left one comment.

File: plugins/woocommerce/client/admin/config/core.json (hunk 1 above)
Comment (retrofox) [reply plugins/woocommerce/client/admin/config/core.json]: 😱

Comment (retrofox) [reply plugins/woocommerce/client/admin/config/core.json]: addressed https://github.com/woocommerce/woocommerce/pull/43006/commits/d335f4fc0baa2adc1e1db530372cf0df0b805b8a

**Review by mdperez86 — APPROVED** [review]
Awesome @retrofox nice job!!!

**Pull request merged**